src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Stream.cs

```diff
@@ -325,35 +325,38 @@ public static partial class JsonSerializer
                         try
                         {
                             isFinalBlock = WriteCore(converter, writer, value, options, ref state);
+                            await bufferWriter.WriteToStreamAsync(utf8Json, cancellationToken).ConfigureAwait(false);
+                            bufferWriter.Clear();
```

Comment:
I don't think `Clear()` will throw, but is fine as-is here. The `PooledByteBufferWriter.Dispose()` will clear it in the end, so it doesn't need to be in the `finally` either.

Reply:
You're right, but moving it elsewhere felt weird from a readability perspective.